tensorflow/python/keras/layers/embeddings.py

```diff
@@ -188,7 +188,7 @@ def compute_output_shape(self, input_shape):
 
   def call(self, inputs):
     dtype = K.dtype(inputs)
-    if dtype != 'int32' and dtype != 'int64':
+    if dtype != 'int32' or dtype != 'int64':
```

Comment:
I don't think this change is correct. The current code is trying to cast the dtype to int32 only if the dtype is not int32 and not int64, eg in the case of float32. More specifically, we don't want to convert the int64 dtype to int32.

See original change in https://github.com/tensorflow/tensorflow/commit/ab02bce13e49fbd001c6db241d213dc2886a5792.